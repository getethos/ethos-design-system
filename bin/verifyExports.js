const index = require('../src/components/index')
const findAllFilesInDirectories = require('./helpers/findAllFilesInDirectories')
const path = require('path')

function verifyExports(files = nonspecJsFiles()) {
  return files
}

export default verifyExports

function nonspecJsFiles() {
  const jsFilesWithSpec = findAllFilesInDirectories(['src/components'], '.js')
  const jsFiles = [...jsFilesWithSpec].filter((file) => {
    return !(
      file.includes('spec') ||
      file.includes('test') ||
      file.includes('index')
    )
  })

  const componentsFromFiles = jsFiles.map((file) => path.basename(file, '.js'))

  const componentsFromIndexImport = Object.entries(index).map(
    (component) => component[0]
  )

  var uniques = componentsFromFiles.filter(
    (obj) => componentsFromIndexImport.indexOf(obj) == -1
  )

  return uniques
}
