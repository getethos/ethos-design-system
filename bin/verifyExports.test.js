import verifyExports from './verifyExports'

describe('Index exports,  ', () => {
  it('if a EDS component was recently added or removed, fix index then update snapshots', () => {
    expect(verifyExports()).toMatchSnapshot()
  })
})
