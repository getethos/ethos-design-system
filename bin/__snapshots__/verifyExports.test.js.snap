// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Index exports,   if a EDS component was recently added or removed, fix index then update snapshots 1`] = `
Array [
  "OptionButton",
  "Colors",
  "FormDynamic",
  "FormPrefilled",
  "Images",
  "InputLabel",
  "RadioButtons",
  "Type",
  "FancyAnimatedLogo",
  "TransformingBurgerButton",
  "ethos-logo-black",
  "ethos-logo-white",
  "fancy-e",
  "fancy-h",
  "fancy-o",
  "fancy-s",
  "fancy-t",
]
`;
